examples/api-tests/src/undo-redo-selectAll.spec.js

```diff
@@ -0,0 +1,188 @@
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// @ts-check
+describe('Undo, Redo and Select All', function () {
+    this.timeout(5000);
+
+    const { assert } = chai;
+
+    const Uri = require('@theia/core/lib/common/uri');
+    const { animationFrame } = require('@theia/core/lib/browser/browser');
+    const { DisposableCollection } = require('@theia/core/lib/common/disposable');
+    const { CommonCommands } = require('@theia/core/lib/browser/common-frontend-contribution');
+    const { EditorManager } = require('@theia/editor/lib/browser/editor-manager');
+    const { WorkspaceService } = require('@theia/workspace/lib/browser/workspace-service');
+    const { CommandRegistry } = require('@theia/core/lib/common/command');
+    const { KeybindingRegistry } = require('@theia/core/lib/browser/keybinding');
+    const { FileNavigatorContribution } = require('@theia/navigator/lib/browser/navigator-contribution');
+    const { ApplicationShell } = require('@theia/core/lib/browser/shell/application-shell');
+    const { MonacoEditor } = require('@theia/monaco/lib/browser/monaco-editor');
+    const { ScmContribution } = require('@theia/scm/lib/browser/scm-contribution');
+
+    const container = window.theia.container;
+    const editorManager = container.get(EditorManager);
+    const workspaceService = container.get(WorkspaceService);
+    const commands = container.get(CommandRegistry);
+    const keybindings = container.get(KeybindingRegistry);
+    const navigatorContribution = container.get(FileNavigatorContribution);
+    const shell = container.get(ApplicationShell);
+    const scmContribution = container.get(ScmContribution);
+
+    const rootUri = new Uri.default(workspaceService.tryGetRoots()[0].uri);
+    const fileUri = rootUri.resolve('webpack.config.js');
+
+    const toTearDown = new DisposableCollection();
+
+    /**
+     * @template T
+     * @param {() => Promise<T> | T} condition
+     * @returns {Promise<T>}
+     */
+    function waitForAnimation(condition) {
+        return new Promise(async (resolve, dispose) => {
+            toTearDown.push({ dispose });
+            do {
+                await animationFrame();
+            } while (!condition());
+            resolve();
+        });
+    }
+
+    before(() => {
+        shell.leftPanelHandler.collapse();
+    });
+
+    beforeEach(async function () {
+        await scmContribution.closeView();
+        await navigatorContribution.closeView();
+        await editorManager.closeAll({ save: false });
+    });
+
+    afterEach(async () => {
+        toTearDown.dispose();
+        await scmContribution.closeView();
+        await navigatorContribution.closeView();
+        await editorManager.closeAll({ save: false });
```

Comment:
During writing tests or running all of them together not everything always goes well and bogus termination of one test can break other tests. If it would be `beforeEach` should not be necessary.